src/lib/rust.ts

```diff
@@ -300,13 +305,36 @@ export const ensureCargoDependency = async ({
     });
 
   try {
-    if (process.env.CI === "true") {
+    if (
+      force ||
+      ![MARINE_CARGO_DEPENDENCY, MREPL_CARGO_DEPENDENCY].includes(name)
+    ) {
+      throw new Error("Install using cargo");
+    }
+
+    if (["darwin", "linux"].includes(platform()) && arch() === "x64") {
       await downloadPrebuiltCargoDependencies(name, version, dependencyPath);
     } else {
-      throw new Error("Not in CI");
+      throw new Error("Unsupported platform. Use cargo");
+    }
+
+    try {
+      // check binary is working
+      const helpText = await execPromise({
+        command: join(dependencyPath, "bin", name),
+        args: ["--help"],
+      });
+
+      if (!helpText.includes(version)) {
+        throw new Error("Version mismatch");
```

Comment:
This error lacks a lot of information. I don't see how user/dev/me can address it.

What software has version mismatch? Linux? Chrome? What versions? Version 1 != version 2? What was expected, and what did you get?